### PR TITLE
[wasm] Support parameters of type Enum with icalls

### LINKED ISSUE #61277: [wasm] Parameters of type Enum not supported with icalls
Based on https://github.com/dotnet/runtime/issues/61053#issuecomment-957227359:

Fails for `[System.Private.CoreLib] System.Enum::InternalHasFlag`:

```
 The "IcallTableGenerator" task failed unexpectedly. [C:\uselessStuff\GameBuddy\src\Client\GameBuddy.Client.csproj]
 System.NotImplementedException: System.Enum [C:\uselessStuff\GameBuddy\src\Client\GameBuddy.Client.csproj]
    at IcallTableGenerator.AppendType(StringBuilder sb, Type t) [C:\uselessStuff\GameBuddy\src\Client\GameBuddy.Client.csproj]      
    at IcallTableGenerator.ProcessType(Type type) [C:\uselessStuff\GameBuddy\src\Client\GameBuddy.Client.csproj]
    at IcallTableGenerator.GenIcallTable(String runtimeIcallTableFile, String[] assemblies) [C:\uselessStuff\GameBuddy\src\Client\GameBuddy.Client.csproj]
    at IcallTableGenerator.Execute() [C:\uselessStuff\GameBuddy\src\Client\GameBuddy.Client.csproj]
    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\uselessStuff\GameBuddy\src\Client\GameBuddy.Client.csproj]
    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [C:\uselessStuff\GameBuddy\src\Client\GameBuddy.Client.csproj]
```

cc @lambdageek @vargaz

### PR DESCRIPTION
**Pull request by maraf**
Support for reading signatures for icalls with overloaded parameters containing enums.

Fixes https://github.com/dotnet/runtime/issues/61277

### CODE REVIEW

File: src/tests/BuildWasmApps/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs [hunk 1]
```diff
@@ -115,6 +116,120 @@ public static int Main()
...
+                    PInvokeOutputPath="$(WasmPInvokeTablePath)"
+                    RuntimeIcallTableFile="$(WasmRuntimeICallTablePath)"
+                    InterpToNativeOutputPath="$(WasmInterpToNativeTablePath)">
+                    <Output TaskParameter="FileWrites" ItemName="FileWrites" />
+                  </ManagedToNativeGenerator>
+                </Target>
+            </Project>
+            """;
+
+            string AddAssembly(string name) => $"<WasmPInvokeAssembly Include=\"{Path.Combine(libraryDir, "bin", "Debug", "net7.0", "browser-wasm", name + ".dll")}\" />";
```
Comment (radical) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs:197]: `net7.0` -> `BuildTestBase.DefaultTargetFramework`

File: src/tests/BuildWasmApps/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs [hunk 2]
```diff
@@ -115,6 +116,120 @@ public static int Main()
...
+                    InitProject: () =>
+                    {
+                        File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), code);
+                    },
+                    Publish: false,
+                    DotnetWasmFromRuntimePack: false,
+                    AssertAppBundle: false
+                )
+            );
+
```
Comment (radical) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs:162]:
```suggestion
```

**Comment by maraf** [comment]
@radical I guess that the packs path is not correct on linux

```
error MSB4062: The "ManagedToNativeGenerator" task could not be loaded from the assembly /datadisks/disk1/work/B31A0A2B/w/B96F09C5/e/snfgt5dd.gketasks/./dotnet-workload/packs/Microsoft.NET.Runtime.WebAssembly.Sdk/7.0.0-ci/tasks/net7.0/WasmAppBuilder.dll. Could not load file or assembly '/datadisks/disk1/work/B31A0A2B/w/B96F09C5/e/snfgt5dd.gketasks/dotnet-workload/packs/Microsoft.NET.Runtime.WebAssembly.Sdk/7.0.0-ci/tasks/net7.0/WasmAppBuilder.dll'. The system cannot find the file specified.
error MSB4062:  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.
```

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-73817-merge-807125bd1ba34ecbb8/Workloads-.PInvokeTableGeneratorTests/1/console.b273b450.log?helixlogtype=result

**Review by radical — APPROVED** [review]

**Comment by radical** [comment]
`Libraries Test Run release coreclr Linux_musl x64 Debug` failure is https://github.com/dotnet/runtime/issues/73688 .

**Pull request merged**